Improve "on this page" & github links sidebar

lena-larionova opened the pull request:
### Summary and reasons

Improvements to the right-side "On this page" navigation:
* If sidebar is open, toggle is always visible. This fixes the issue of having a hidden sidebar creating space on pages where it shouldn't. 
  * For example: https://docs.konghq.com/gateway/2.7.x/admin-api/ - with the sidebar and toggle completely hidden, there is no way to close the sidebar, and therefore no way to make the text expand to fill the page. You have to switch to some other page, toggle the sidebar, and then return here for it to work. This change fixes that.
* Github links are now always visible, even if the page has no headers.
* Hide 'on this page' if there are no headers
  * Current behavior: even if there are no h2 or h3 elements on the page, an "On this page" link appears (for example: https://docs.konghq.com/gateway/2.7.x/developer-portal/).

### Testing
Check pages without a ToC, they should have github links and a panel toggle:
https://deploy-preview-3624--kongdocs.netlify.app/gateway/2.7.x/admin-api/
https://deploy-preview-3624--kongdocs.netlify.app/gateway/2.7.x/developer-portal

Check any other pages to make sure that the "On this page" title for the nav displays correctly, for example:
https://deploy-preview-3624--kongdocs.netlify.app/gateway/
https://deploy-preview-3624--kongdocs.netlify.app/gateway/changelog/
https://deploy-preview-3624--kongdocs.netlify.app/hub/kong-inc/application-registration/

falondarville commented:
LGTM. I found some some awkward spacing above the menu on around 1240px.

![Screen Shot 2022-02-04 at 9 05 16 AM](https://user-images.githubusercontent.com/22301405/152571513-268db3bc-aad8-4b17-bd6a-d6fcbab256ff.png)

lena-larionova commented:
Changes made after review:
- Fixed margin on the sidebar
- Ensured that toggles for sidebar are always showing and don't disappear at some sizes
- Removed `toc:false` from latest versions of docs that don't need it 
- Removed `toc:false` from markdown guide because we don't really want people using it. We can still set it if needed in some edge cases (eg Kong Admin API doc, and that's pretty much it), but it shouldn't be used randomly.

falondarville commented:
Edits look good.
